From e85d3df03ca8c226364df61e3dbeee6ccddd5fe1 Mon Sep 17 00:00:00 2001
From: Axel Huebl <axel.huebl@plasma.ninja>
Date: Tue, 11 Jun 2024 17:15:37 -0700
Subject: [PATCH] Fix AppleClang: Stock `aligned_alloc`

Compiling heFFTe 2.4.0 with AppleClang 16.0.6 on macOS 13.4 raises:
```
include/stock_fft/heffte_stock_allocator.h:43:46: error: use of undeclared identifier 'aligned_alloc'
            return reinterpret_cast<pointer>(aligned_alloc(alignof(F), n*sizeof(F)));
                                             ^
```

This adds the missing include and uses the C++ variant of the
function.
---
 include/stock_fft/heffte_stock_allocator.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/stock_fft/heffte_stock_allocator.h b/include/stock_fft/heffte_stock_allocator.h
index d8d69df..6123513 100644
--- a/include/stock_fft/heffte_stock_allocator.h
+++ b/include/stock_fft/heffte_stock_allocator.h
@@ -2,6 +2,7 @@
 #define HEFFTE_STOCK_ALLOCATOR_H
 
 #include <memory>
+#include <stdlib.h>
 #include <vector>
 
 #include "heffte_stock_complex.h"
